tests/rptest/tests/usage_test.py

```diff
@@ -28,56 +29,162 @@
 from rptest.utils.functional import flat_map
 
 
+class UsageWindow:
+    def __init__(self, begin, end, is_open, bytes_sent, bytes_received,
+                 bytes_in_cloud_storage):
+        self.begin = begin
+        self.end = end
+        self.is_open = is_open
+        self.bytes_sent = bytes_sent
+        self.bytes_received = bytes_received
+        self.bytes_in_cloud_storage = bytes_in_cloud_storage
+
+    def __repr__(self):
+        r = {
+            'begin': self.begin,
+            'end': self.end,
+            'is_open': self.is_open,
+            'bytes_sent': self.bytes_sent,
+            'bytes_received': self.bytes_received,
+            'bytes_in_cloud_storage': self.bytes_in_cloud_storage
+        }
+        return str(r)
+
+
+def parse_usage_response(usage_response):
+    def parse_usage_window(e):
+        return UsageWindow(e['begin_timestamp'], e['end_timestamp'], e['open'],
+                           e['kafka_bytes_sent_count'],
+                           e['kafka_bytes_received_count'],
+                           e['cloud_storage_bytes_gauge'])
+
+    return [parse_usage_window(e) for e in usage_response]
+
+
+def assert_usage_consistency(a, b):
+    """
+    This method returns true if windows with matching timestamps contain the same values
+
+    a: List[UsageWindow]
+    b: List[UsageWindow]
+
+    Windows from list b are assumed to be from a newer response, so this method will verify
+    that all items from b, if they exist in a, are equivalent.
+
+    One excepton if the window is considered an 'open' window, there would be at max 1
+    per request, if there is a matching open window in both inputs, their values
+    are checked so that the value of b's open window would be >= the value in a's.
+    """
+    def group_windows(ws):
+        return {(x.begin, x.end): x for x in ws}
+
+    def compare(x, y, fn):
+        return fn(x.bytes_sent, y.bytes_sent) and fn(
+            x.bytes_received, y.bytes_received) and fn(
+                x.bytes_in_cloud_storage, y.bytes_in_cloud_storage)
+
+    # Windows from group 'a' are expected to be from a previous request
+    grp_a = group_windows(a)
+    for x in b:
+        key = (x.begin, x.end)
+        a_value = grp_a.get(key, None)
+        if a_value is not None:
+            fn = operator.le if a_value.is_open else operator.eq
+            assert compare(a_value, x, fn), f"a: {str(a_value)} b:{str(x)}"
```

Comment:
`test_usage_metrics_collection` failed here in CI with an error I don't really understand:
```
a: {'begin': 1690268117, 'end': 1690268118, 'is_open': False, 'bytes_sent': 0, 'bytes_received': 0, 'bytes_in_cloud_storage': -1} b:{'begin': 1690268117, 'end': 1690268118, 'is_open': False, 'bytes_sent': 0, 'bytes_received': 0, 'bytes_in_cloud_storage': 0}
```

Why would `bytes_in_cloud_storage` change for a closed bucket?

Reply:
Looks like a test error, its possible its comparing results from two different nodes, going to investigate that